src/Common/src/Interop/Unix/System.Security.Cryptography.Native/Interop.ManagedSslBio.cs

```diff
@@ -0,0 +1,66 @@
+using System;
+using System.Runtime.InteropServices;
+using Microsoft.Win32.SafeHandles;
+using System.Diagnostics;
+
+internal static partial class Interop
+{
+    internal static partial class Crypto
+    {
+        internal static class ManagedSslBio
+        {
+            private unsafe static ReadDelegate s_readDelegate;
+            private unsafe static WriteDelegate s_writeDelegate;
+
+            static ManagedSslBio() => Initialize();
+
+            internal static SafeBioHandle CreateManagedSslBio() => Crypto.CreateManagedSslBio();
+            
+            private unsafe static void Initialize()
+            {
+                s_writeDelegate = Write;
+                s_readDelegate = Read;
+                Crypto.InitManagedSslBioMethod(s_writeDelegate, s_readDelegate);
+            }
+
+            internal static void BioSetGCHandle(SafeBioHandle bio, GCHandle handle)
+            {
+                IntPtr pointer;
+                if (handle.IsAllocated)
+                {
+                    pointer = GCHandle.ToIntPtr(handle);
+                }
+                else
+                {
+                    pointer = IntPtr.Zero;
+                }
+
+                Crypto.BioSetAppData(bio, pointer);
+            }
+
+            private static unsafe int Write(IntPtr bio, void* input, int size, IntPtr data)
+            {
+                GCHandle handle = GCHandle.FromIntPtr(data);
+
+                if (handle.IsAllocated && handle.Target is SafeSslHandle.WriteBioBuffer buffer)
```

Comment:
I was more worried that if there were cases it was trying to catch, e.g. it getting free'd after some call getting initiated, that there would be other lurking problems.  I'm fine with it remaining for safety's sake, but it'd probably be good to also add an assert.